sys/kern/kern_descrip.c

```diff
@@ -1485,8 +1508,10 @@ kern_close_range(struct thread *td, int flags, u_int lowfd, u_int highfd)
 		return (EINVAL);
 	}
 
-	if ((flags & CLOSE_RANGE_CLOEXEC) != 0)
-		return (close_range_cloexec(td, lowfd, highfd));
+	if ((flags & (CLOSE_RANGE_CLOEXEC | CLOSE_RANGE_CLOFORK)) != 0)
+		return (close_range_flags(td, lowfd, highfd,
+		    ((flags & CLOSE_RANGE_CLOEXEC) != 0 ? UF_EXCLOSE : 0) |
+		    ((flags & CLOSE_RANGE_CLOFORK) != 0 ? UF_FOCLOSE : 0)));
```

Comment:
I think this conversion is better done in `close_range_flags()`? It doesn't matter too much right now, but if there are more callers in the future it would be silly to duplicate this conversion.

Reply:
Fixed. Looks much better now.